### PR TITLE
Refactor integration tests

### PR DESCRIPTION
**Pull request by dandavison**
This is pure refactoring of the integration tests, done in preparation for implementation of Workflow Update in https://github.com/temporalio/sdk-typescript/pull/1277.

### CODE REVIEW

**Review by bergundy — APPROVED** [review]

**Comment by bergundy** [comment]
Please fix the lint warnings before merging.

File: packages/test/src/integration-tests-old.ts
Comment (mjameswh) [inline packages/test/src/integration-tests-old.ts:1]: Can you please add a comment early in this file that explains how these tests are "old" (ie. they are still valid and up-to-date, but need to be converted to the new testing approach).

File: packages/test/src/integration-tests/helpers.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  DefaultLogger,
+  LogLevel,
+  Runtime,
+  WorkerOptions,
+  WorkflowBundle,
+  bundleWorkflowCode,
+} from '@temporalio/worker';
+import * as workflow from '@temporalio/workflow';
+import { ConnectionInjectorInterceptor } from '../activities/interceptors';
+import { Worker, test as anyTest, bundlerOptions } from '../helpers';
```
Comment (mjameswh) [inline packages/test/src/integration-tests/helpers.ts:19]: Would it be difficult/disruptive to merge the content of this file into `../helpers.ts`?

**Review by mjameswh — APPROVED** [review]
I'm just a bit uncomfortable with file names (`-old` and `helpers`). I gave some details inline.

Not blocking however. Feel free to merge as is if you prefer. These can be addressed at a later point.

**Comment by dandavison** [comment]
Thanks for reviewing. The file name in this PR I am most uncomfortable with is `test-integration-workflows.ts => integration-tests/test-misc.ts`! Please rename that if you can think of a better name. But I think the `-old` suffix is helpful. My objectives with this PR were to eliminate duplication of helper boilerplate, make it easy to discern which are our new-style integration tests, and which our old, while allowing the new-style integration tests to be spread over multiple files (e.g. for the coming update tests, which are in a few different files due to the use of difference interceptors). We don't have to keep the new-style ones in a directory; we could have `test-integration-*.ts` and `test-integration-old-*.ts`.

File: packages/test/src/integration-tests/helpers.ts (hunk 1 above)
Comment (dandavison) [reply packages/test/src/integration-tests/helpers.ts:19]: That sounds good, I'm happy for any further renaming/reorganizing tweaks to be done here; I just wanted to start getting things in shape while adding new tests for Update.

**Pull request merged**